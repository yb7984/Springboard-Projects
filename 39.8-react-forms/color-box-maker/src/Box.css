.Box{
    display: flex;
    flex-wrap: nowrap;
    padding: 10px;
    margin: 10px;
    position: relative;
}

.Box-box{
    border-radius: 10px;
}
.Box-btn{
    padding: 5px;
    width: 30px;
    height: 30px;
    color: red;
    position: relative;
    left: -30px;
}