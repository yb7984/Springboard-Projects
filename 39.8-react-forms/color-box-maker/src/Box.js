import "./Box.css";

const Box = ({ color, width, height, id, handlDelete }) => {
    return (
        <div className="Box">
            <div className="Box-box" data-testid={id} style={{ backgroundColor: color, width: `${width}px`, height: `${height}px` }}></div>
            <button className="Box-btn" data-testid={`del-${id}`} onClick={() => { handlDelete(id) }}>X</button>
        </div >)
}


export default Box;