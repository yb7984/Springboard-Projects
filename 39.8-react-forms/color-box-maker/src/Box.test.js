import { render } from '@testing-library/react';
import Box from "./Box";

it("renders without crashing", function () {
    render(<Box id="test" width="100" height="100" color="#000000" />);
});


it("matches snapshot, have certain style", function () {
    const { asFragment, queryByTestId } = render(<Box id="test" width="100" height="100" color="#000000" />);
    expect(asFragment()).toMatchSnapshot();

    expect(queryByTestId("test")).toHaveStyle({
        width: "100px",
        height: "100px",
        backgroundColor: "#000000"
    });

});
