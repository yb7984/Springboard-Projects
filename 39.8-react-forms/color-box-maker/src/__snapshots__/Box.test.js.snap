// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot, have certain style 1`] = `
<DocumentFragment>
  <div
    class="Box"
  >
    <div
      class="Box-box"
      data-testid="test"
      style="background-color: rgb(0, 0, 0); width: 100px; height: 100px;"
    />
    <button
      class="Box-btn"
      data-testid="del-test"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
